### PR TITLE
remove asan workaround

### PR DESCRIPTION
**Pull request by lizan**
Signed-off-by: Lizan Zhou <lizan@tetrate.io>

### CODE REVIEW

**Comment by lizan** [comment]
@PiotrSikora

**Comment by PiotrSikora** [comment]
It ran out of disk space, which is why John had to change `dbg` to `opt` in the first place.

But the issue I have with this change is that you're actually pretty likely to run release builds with ASan in production to catch errors, so removing this configuration from the CI doesn't actually benefits anybody.

**Comment by lizan** [comment]
> It ran out of disk space, which is why John had to change `dbg` to `opt` in the first place.
> 
> But the issue I have with this change is that you're actually pretty likely to run release builds with ASan in production to catch errors, so removing this configuration from the CI doesn't actually benefits anybody.

It is acutally OOM, I'll dig a bit more. We can do -O1, which is clang docs suggests.

**Review by alyssawilk — APPROVED** [review]

File: .bazelrc [hunk 1]
```diff
@@ -62,6 +62,9 @@ build:asan --copt -D__SANITIZE_ADDRESS__
 build:asan --test_env=ASAN_OPTIONS=handle_abort=1:allow_addr2line=true:check_initialization_order=true:strict_init_order=true:detect_odr_violation=1
 build:asan --test_env=UBSAN_OPTIONS=halt_on_error=true:print_stacktrace=1
 build:asan --test_env=ASAN_SYMBOLIZER_PATH
+# ASAN needs -O1 to get reasonable performance.
+build:asan --copt -O1
+build:asan --copt -foptimize-sibling-calls
```
Comment (PiotrSikora) [inline .bazelrc:67]: If you're following https://releases.llvm.org/10.0.0/tools/clang/docs/AddressSanitizer.html, shouldn't this be `-fno-optimize-sibling-calls`?

Comment (lizan) [reply .bazelrc:67]: :facepalm: https://github.com/envoyproxy/envoy/pull/13691

**Pull request merged**